### PR TITLE
Feature/income expense view

### PR DESCRIPTION
**Pull request by E-LeeSchaffer**

### CODE REVIEW

File: components/AccountBalance.js [hunk 1]
```diff
@@ -2,37 +2,90 @@ import { useEffect, useState } from "react";
...
+    const incomeTotal = transactions
+      .filter((transaction) => transaction.type === "income")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    const expenseTotal = transactions
+      .filter((transaction) => transaction.type === "expense")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    setCurrentBalance(incomeTotal - expenseTotal);
   }, [transactions]);
```
Comment (F-Kirchhoff) [inline components/AccountBalance.js:22]: this useEffect is not needed. you can calculate incomeTotal and expenseTotal in the component directly.

File: components/AccountBalance.js [hunk 2]
```diff
@@ -2,37 +2,90 @@ import { useEffect, useState } from "react";
...
 
   const currentBalanceType = currentBalance < 0 ? "expense" : "income";
 
+  function getTitle() {
+    if (filteredAccountType === "income") return "Total Income";
+    if (filteredAccountType === "expense") return "Total Expense";
+    if (filteredAccountType === "total") return "Total sum";
+    return "Current Balance";
+  }
```
Comment (F-Kirchhoff) [inline components/AccountBalance.js:31]: nice solution! there is one that is a bit more concise using an object: 
```js
const titleMap = {
  income: "Total Income",
  expense: "Total Expense",
  total: "Total Sum"
}

const title = titleMap[filteredAccountType] || "Current Balance"
```
Feel free to keep your solution or switch to this one.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Eun-Rae,
I summarised the things that we could not discuss today here. If you get stuck please reach out 😊

File: components/AccountBalance.js (hunk 2 above)
Comment (E-LeeSchaffer) [reply components/AccountBalance.js:31]: I really like this solution! I have taken this solution.

File: components/AccountBalance.js [hunk 3]
```diff
@@ -46,10 +35,19 @@ export default function AccountBalance({
...
+  const amountMap = {
+    income: income,
+    expense: expense,
+    total: total,
+  };
+
+  const displayAmount =
+    filteredTransactionType in amountMap
+      ? amountMap[filteredTransactionType]
+      : currentBalance;
```
[the remark this answers is not in the document]
Comment (E-LeeSchaffer) [reply components/AccountBalance.js:47]: Yes, I have deleted line 20. Thanks.

**Pull request merged**